torch/csrc/distributed/c10d/ProcessGroupUCC.cpp

```diff
@@ -128,11 +129,58 @@ std::map<std::string, std::string> torch_ucc_envs_map = {
     {"TORCH_UCC_ENABLE_COMMS_LOGGER", "0"},
 };
 
+// trim implementation borrowed from https://stackoverflow.com/a/17976541
+inline c10::string_view trim(c10::string_view s) {
+  auto wsfront = std::find_if_not(
+      s.begin(), s.end(), [](int c) { return std::isspace(c); });
+  auto wsback = std::find_if_not(s.rbegin(), s.rend(), [](int c) {
+                  return std::isspace(c);
+                }).base();
+  return (wsback <= wsfront ? "" : s.substr(wsfront - s.begin(), wsback - wsfront));
+}
+
+std::unordered_set<OpType> parse_blocking_wait(std::string s) {
+  const static std::unordered_map<std::string, OpType> str2op = {
+      {"allgather", OpType::ALLGATHER},
+      {"allgather_base", OpType::_ALLGATHER_BASE},
+      {"allreduce", OpType::ALLREDUCE},
+      {"alltoall_base", OpType::ALLTOALL_BASE},
+      {"broadcast", OpType::BROADCAST},
+      {"gather", OpType::GATHER},
+      {"reduce", OpType::REDUCE},
+      {"reduce_scatter", OpType::REDUCE_SCATTER},
+      {"scatter", OpType::SCATTER},
+      {"send", OpType::SEND},
+      {"recv", OpType::RECV},
+  };
+  std::unordered_set<OpType> result;
+  c10::string_view sv = trim(s);
+  if (sv == "0") {
+    return result;
+  }
+  if (sv == "1" || sv == "all") {
+    result.reserve(str2op.size());
+    for (auto item : str2op) {
+      result.insert(item.second);
+    }
+    return result;
+  }
+  while (!sv.empty()) {
+    const auto end_pos = sv.find_first_of(',');
+    const auto token = trim(sv.substr(0, end_pos));
+    const auto it = str2op.find(std::string(token));
+    TORCH_CHECK(it != str2op.end(), "Invalid blocking wait option: '", token);
+    result.insert(it->second);
+    sv = (end_pos != c10::string_view::npos) ? sv.substr(end_pos + 1) : "";
+  }
+  return result;
+}
+
```

Comment:
Would it look cleaner if all the new code above for parsing env is moved to UCCUtils.cpp?

Reply:
Would be nice to have `tolower()` support.

Reply:
`tolower` added. I can not move all of them, because `torch_ucc_envs_map` is used in other places in this file, and `parse_blocking_wait` uses `OpType`, which is defined in `ProcessGroupUCC.hpp`. But I moved the inline functions, `trim` and `tolower` into `UCCUtils.hpp`

Reply:
done